esmvalcore/_citation.py

```diff
@@ -0,0 +1,229 @@
+        except IOError:
+            logger.info('Error in receiving the CMIP json file')
+    return json_data
+
+
+def _json_to_bibtex(data):
+    """Make a bibtex entry from CMIP6 Data Citation json data."""
+    url = 'url not found'
+    title = data.get('titles', ['title not found'])[0]
+    publisher = data.get('publisher', 'publisher not found')
+    year = data.get('publicationYear', 'publicationYear not found')
+    authors = 'creators not found'
+    doi = 'doi not found'
+
+    author_list = []
+    if data.get('creators', False):
+        author_list = [
+            item.get('creatorName', '') for item in data['creators']
+        ]
+    if author_list:
+        if author_list[0] == author_list[-1]:
+            authors = author_list[0]
+            if not authors:
+                authors = 'creatorName not found'
+        else:
+            authors = ' and '.join(author_list)
+
+    if data.get('identifier', False):
+        doi = data.get('identifier').get('id', 'doi not found')
+        url = f'https://doi.org/{doi}'
+
+    bibtex_entry = (
+        f'{"@misc{"}{url},\n\t'
+        f'url = {{{url}}},\n\t'
+        f'title = {{{title}}},\n\t'
+        f'publisher = {{{publisher}}},\n\t'
+        f'year = {year},\n\t'
+        f'author = {{{authors}}},\n\t'
+        f'doi = {{{doi}}},\n'
+        f'{"}"}\n'
+    )
+    return bibtex_entry
+
+
+def _collect_bibtex_citation(tag):
+    """Collect information from bibtex files."""
+    bibtex_file = REFERENCES_PATH / f'{tag}.bibtex'
+    if bibtex_file.is_file():
+        entry = bibtex_file.read_text()
+    else:
+        logger.info(
+            'The reference file %s does not exist.', bibtex_file
+        )
+        entry = ''
+    return entry
+
+
+def _collect_cmip_citation(json_url):
+    """Collect information from CMIP6 Data Citation Service."""
+    json_data = _get_response(json_url)
+    if json_data:
+        bibtex_entry = _json_to_bibtex(json_data)
+    else:
+        logger.info('Invalid json link %s', json_url)
+        bibtex_entry = False
+    return bibtex_entry
+
+
+def _make_url_prefix(attribute):
+    """Make url prefix based on CMIP6 Data Citation Service."""
+    # the order of keys is important
+    localpart = {
+        'mip_era': '',
+        'activity_id': '',
+        'institution_id': '',
+        'source_id': '',
+        'experiment_id': '',
+    }
+    for key, value in attribute:
+        if key.localpart in localpart:
+            localpart[key.localpart] = value
+    url_prefix = '.'.join(localpart.values())
+    return url_prefix
+
+
+def _make_json_url(url_prefix):
+    """Make json url based on CMIP6 Data Citation Service."""
+    json_url = f'{CMIP6_URL_STEM}/cerarest/exportcmip6?input={url_prefix}'
+    return json_url
+
+
+def _make_info_url(url_prefix):
+    """Make info url based on CMIP6 Data Citation Service."""
+    info_url = f'{CMIP6_URL_STEM}/cmip6?input=CMIP6.{url_prefix}'
+    return info_url
+
+
+def cite_tag_value(tags):
```

Comment:
This function should be moved to ` esmvaltool/cmorizers/obs/utilities.py` in the ESMValTool repository, because that is the only place where it is used.

Provided that the information is available, the return value should be a string looking like
```
"Author X., et al., Journal Y, doi:xyz, year."
```
instead of the complete contents of the bibtex file. I think we discussed this, my apologies if that was not clear.

Reply:
Discussed this, will use only DOI for now